### PR TITLE
Task as/j15 frontend part episode page

### PR DESCRIPTION
**Pull request by qwiz1**
episodePg_windowWidth=2560
episodePg_windowWidth=1440

### CODE REVIEW

File: frontend/src/store/episodepage/reducer.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+type State = {
+  dataStatus: DataStatus;
+  episode: Episode | null;
+  episodes: Episode[];
+};
+
+const initialState: State = {
+  dataStatus: DataStatus.IDLE,
+  episode: null,
+  episodes: [],
```
Comment (ghost) [inline frontend/src/store/episodepage/reducer.ts]:
```suggestion
};

const initialState: State = {
  dataStatus: DataStatus.IDLE,
  episode: null,
```

File: frontend/src/store/episodepage/common.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+enum ActionType {
+  LOAD_EPISODE = 'episodes/load-episode',
+  LOAD_EPISODES = 'episodes/load-episodes',
```
Comment (ghost) [inline frontend/src/store/episodepage/common.ts]:
```suggestion
```

File: frontend/src/store/episodepage/actions.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  return episode;
+});
+
+const loadEpisodes = createAsyncThunk<Episode[], undefined, AsyncThunkConfig>
+(ActionType.LOAD_EPISODES, async (_args, { extra }) => {
+  const { episodeApi } = extra;
+  const episodes = await episodeApi.getAll();
+
+  return episodes;
+});
```
Comment (ghost) [inline frontend/src/store/episodepage/actions.ts]:
```suggestion
```

File: frontend/src/store/root-reducer.ts [hunk 4]
```diff
@@ -1,10 +1,12 @@
 import { reducer as auth } from './auth/reducer';
 import { reducer as homepage } from './homepage/reducer';
 import { reducer as toastrReducer } from 'react-redux-toastr';
+import { reducer as episodepage } from './episodepage/reducer';
```
Comment (ghost) [inline frontend/src/store/root-reducer.ts]:
```suggestion
import { reducer as episode } from './episode/reducer';
```

File: frontend/src/store/episodepage/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  builder.addCase(loadEpisodes.pending, (state) => {
+    state.dataStatus = DataStatus.PENDING;
+  });
+  builder.addCase(loadEpisodes.fulfilled, (state, action) => {
+    state.dataStatus = DataStatus.FULFILLED;
+    state.episodes = action.payload;
+  });
+  builder.addCase(loadEpisodes.rejected, (state) => {
+    state.dataStatus = DataStatus.REJECTED;
+  });
```
Comment (ghost) [inline frontend/src/store/episodepage/reducer.ts]:
```suggestion
```

File: frontend/src/store/actions.ts [hunk 6]
```diff
@@ -1,2 +1,3 @@
 export * as auth from './auth/actions';
 export * as homepage from './homepage/actions';
+export * as episodepage from './episodepage/actions';
```
Comment (ghost) [inline frontend/src/store/actions.ts]:
```suggestion
export * as episode from './episodepage/actions';
```

File: frontend/src/components/common/episode/episode.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  const hasEpisode = Boolean(episode);
+
+  const episodeMock = {
+    podcastsGroup: 'English podcasts',
+    img: 'https://i.pinimg.com/474x/19/28/de/1928deaa3f4ba47822a5c1723f4b930f.jpg',
+    description: 'Aliens send Rick, Morty and Jerry into an alternate reality, and Rick tries to get them out as oblivious Jerry pitches a marketing slogan for apples.',
+  };
+
+  return (
+    <section className={styles.episode}>
```
Comment (ghost) [inline frontend/src/components/common/episode/episode.tsx]:
```suggestion
    <div className={styles.episode}>
```

File: frontend/src/components/common/episode/episode.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      <section className={styles.episodeHeader}>
+        {hasEpisode ? (
+          <>
+            <div className={styles.description}>
+              <h5>{episodeMock.podcastsGroup}</h5>
```
Comment (ghost) [inline frontend/src/components/common/episode/episode.tsx]:
```suggestion
              <h1>{episodeMock.podcastsGroup}</h1>
```

File: frontend/src/components/common/episode/episode.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    episode: episodepage.episode,
+  }));
+
+  const hasEpisode = Boolean(episode);
+
+  const episodeMock = {
+    podcastsGroup: 'English podcasts',
+    img: 'https://i.pinimg.com/474x/19/28/de/1928deaa3f4ba47822a5c1723f4b930f.jpg',
+    description: 'Aliens send Rick, Morty and Jerry into an alternate reality, and Rick tries to get them out as oblivious Jerry pitches a marketing slogan for apples.',
+  };
```
Comment (ghost) [inline frontend/src/components/common/episode/episode.tsx]:
```suggestion
```

File: frontend/src/components/common/episode/episode.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+  return (
+    <section className={styles.episode}>
+      <section className={styles.episodeHeader}>
+        {hasEpisode ? (
+          <>
+            <div className={styles.description}>
+              <h5>{episodeMock.podcastsGroup}</h5>
+              <h3>{episode?.name}</h3>
```
Comment (ghost) [inline frontend/src/components/common/episode/episode.tsx]:
```suggestion
```

File: frontend/src/components/common/episode/episode.tsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+              <p>{episodeMock.description}</p>
+            </div>
+            <div className={styles.logo}>
+              <img src={episodeMock.img} alt="logo" />
```
Comment (ghost) [inline frontend/src/components/common/episode/episode.tsx]: always add width, height, loading attributes to `img`

Comment (ghost) [inline frontend/src/components/common/episode/episode.tsx]:
```suggestion
              <img src={episodeMock.img} alt={episode.name} />
```

File: frontend/src/components/common/episode/episode.tsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+            </div>
+          </>
+        ) : (
+          <p>Ups, something went wrong</p>
```
Comment (ghost) [inline frontend/src/components/common/episode/episode.tsx]:
```suggestion
          <h1>There is no such episode</h1>
```

File: frontend/src/components/common/episode/episode.tsx [hunk 13]
```diff
@@ -0,0 +1,50 @@
+import { useAppSelector, useDispatch, useLocation, useEffect } from 'hooks/hooks';
+import { RootState } from 'common/types/types';
+import { episodepage as episodeActions } from 'store/actions';
+import styles from './styles.module.scss';
+
+const Episode: React.FC = () => {
+  const dispatch = useDispatch();
+  const { pathname } = useLocation();
+  const episodeId = Number(pathname.split('/').pop());
```
Comment (ghost) [inline frontend/src/components/common/episode/episode.tsx]: use `useParams`

File: frontend/src/components/common/episode/episode.tsx (hunk 9 above)
Comment (ghost) [reply frontend/src/components/common/episode/episode.tsx]: Take it from the store

File: frontend/src/components/common/episode/episode.tsx (hunk 10 above)
Comment (ghost) [reply frontend/src/components/common/episode/episode.tsx]: don't forget remove styles

File: frontend/src/components/common/episode/episode.tsx [hunk 14]
```diff
@@ -0,0 +1,50 @@
+import { useAppSelector, useDispatch, useLocation, useEffect } from 'hooks/hooks';
```
Comment (yevhenprybytok) [inline frontend/src/components/common/episode/episode.tsx]: It looks like this should be considered as a page, so let's move this to the root level of the components directory

File: frontend/src/components/common/episode/episode.tsx [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+  const episodeMock = {
+    podcastsGroup: 'English podcasts',
+    img: 'https://i.pinimg.com/474x/19/28/de/1928deaa3f4ba47822a5c1723f4b930f.jpg',
+    description: 'Aliens send Rick, Morty and Jerry into an alternate reality, and Rick tries to get them out as oblivious Jerry pitches a marketing slogan for apples.',
+  };
+
+  return (
+    <section className={styles.episode}>
+      <section className={styles.episodeHeader}>
+        {hasEpisode ? (
```
Comment (yevhenprybytok) [inline frontend/src/components/common/episode/episode.tsx]: Ideally, we should not check this here, this episode page should not appear if it's absent

**Review by yevhenprybytok — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/episode/episode.tsx (hunk 11 above)
Comment (yevhenprybytok) [inline frontend/src/components/common/episode/episode.tsx]:
```suggestion
              <img src={episodeMock.img} alt="Episode image" />
```

**Review by yevhenprybytok — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/common.ts [hunk 16]
```diff
@@ -7,6 +7,7 @@ export { default as Header } from './header/header';
 export { default as Checkbox } from './checkbox/checkbox';
 export { default as NotFound } from './not-found/not-found';
 export { default as PrivateRoute } from './private-route/private-route';
+export { default as Episode } from './episode/episode';
```
Comment (ghost) [inline frontend/src/components/common/common.ts]:
```suggestion
```

File: frontend/src/components/common/episode/episode.tsx (hunk 14 above)
Comment (ghost) [inline frontend/src/components/common/episode/episode.tsx]: move to `frontend/src/components/episode/episode.tsx`

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by JuliaDzyuba — DISMISSED** [review]

File: frontend/src/components/common/episode/episode.tsx (hunk 9 above)
Comment (qwiz1) [reply frontend/src/components/common/episode/episode.tsx]: > Take it from the store
Sorry, I didn't understand what exactly is meant and how to implement this. I would be glad if your hint.

File: frontend/src/store/episode/reducer.ts [hunk 17]
```diff
@@ -0,0 +1,29 @@
+import { createReducer } from '@reduxjs/toolkit';
+import { DataStatus } from 'common/enums/enums';
+import { Episode } from 'common/types/types';
+import { loadEpisode } from './actions';
+
+type State = {
+  dataStatus: DataStatus;
+  episodeInfo: Episode | null;
```
Comment (ghost) [inline frontend/src/store/episode/reducer.ts]:
```suggestion
  episode: Episode | null;
```

File: frontend/src/store/episode/reducer.ts [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+const initialState: State = {
+  dataStatus: DataStatus.IDLE,
+  episodeInfo: null,
```
Comment (ghost) [inline frontend/src/store/episode/reducer.ts]:
```suggestion
  episode: null,
```

File: frontend/src/store/episode/common.ts [hunk 19]
```diff
@@ -0,0 +1,5 @@
+enum ActionType {
+  LOAD_EPISODE = 'episodes/load-episode',
```
Comment (ghost) [inline frontend/src/store/episode/common.ts]:
```suggestion
  LOAD_EPISODE = 'episode/load-episode',
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**